grpc/src/main/java/com/linecorp/armeria/server/grpc/DeferredListener.java

```diff
@@ -75,8 +75,23 @@
 
             this.delegate = delegate;
             try {
-                for (Consumer<Listener<I>> task : pendingTasks) {
-                    task.accept(delegate);
+                for (;;) {
+                    final List<Consumer<Listener<I>>> pendingTasks = this.pendingTasks;
+                    if (pendingTasks.isEmpty()) {
+                        break;
+                    }
+
+                    // New pending tasks could be added while invoking pending tasks.
+                    this.pendingTasks = new ArrayList<>();
```

Comment:
The fallback `pendingTasks` might be not frequently used in the real world. Since the problem happens when 1) an event loop is used to serve the service and 2) the service method responds to the request synchronously. On second thought, it would be efficient to minimize the allocation by adding a temporary placeholder.
```java
static List<Consumer<Listener<I>>> TEMPORARY_TASKS = 
        Collections.unmodifiableList(new ArrayList<>());

...
// New pending tasks could be added while invoking pending tasks.
// maybeAddPendingTask() replaces TEMPORARY_TASKS with an appendable one. 
this.pendingTasks = TEMPORARY_TASKS;


private void maybeAddPendingTask(Consumer<ServerCall.Listener<I>> task) {
        if (eventLoop != null && eventLoop.inEventLoop()) {
            addPendingTasks(task);
        } ...
}

private void addPendingTask(task) {
    if (pendingTasks == TEMPORARY_TASKS) {
        pendingTasks = new ArrayList<>();
    }
    pendingTasks.add(task);
}
```

Reply:
Should we keep using `pendingTasks` even if pendingTasks is empty? I think we can run immediately.

Reply:
Fixed